WooCommerce/Classes/ViewRelated/Products/Edit Product/EditCategories/ProductCategoryViewModelBuilder.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import Yosemite
+
+/// Creates `ProductCategoryViewModel` types
+///
+struct ProductCategoryViewModelBuilder {
```

Comment:
I normally like to have these builders as stateless as possible. I think that it's better for the view-model to be source of truth rather than having state in two places where one could get outdated.

What do you think?

Reply:
I don't think the state matters much if it's just used once and not kept forever. 

```swift
let builder = Builder(categories: categories, selectedCategories: selectedCategories)
let viewModels = builder.create()
```

I was more curious about how readable the class will be if there are lesser arguments passed around. 

But this is fine. If it ever happens that we'll have to _mock_ the `Builder` then we can think about it then.

Reply:
👍